addons/sourcemod/scripting/vsh/abilities/ability_rage_gas.sp

```diff
@@ -0,0 +1,191 @@
+		}
+		public get()
+		{
+			return g_flNewSpeed[this.iClient];
+		}
+	}
+	
+	property float flMaxDistance
+	{
+		public set (float flVal)
+		{
+			g_flBombProjectileMaxDistance[this.iClient] = flVal;
+		}
+		public get()
+		{
+			return g_flBombProjectileMaxDistance[this.iClient];
+		}
+	}
+	
+	property float flHeight
+	{
+		public set (float flVal)
+		{
+			g_flBombProjectileHeight[this.iClient] = flVal;
+		}
+		public get()
+		{
+			return g_flBombProjectileHeight[this.iClient];
+		}
+	}
+		
+	public CRageGas(CRageGas ability)
+	{
+		g_flBombProjectileNext[ability.iClient] = 0.0;
+		g_flBombProjectileEnd[ability.iClient] = 0.0;
+		
+		ability.flRate = 1.7;
+		ability.flDuration = 8.0;
+		ability.flMaxDistance = 600.0;
+		ability.flHeight = 700.0;
+	}
+
+	public void OnRage()
+	{
+		g_flBombProjectileNext[this.iClient] = GetGameTime();
+		g_flBombProjectileEnd[this.iClient] = GetGameTime() + this.flDuration;
+		g_flPreviousSpeed[this.iClient] = this.flSpeed;
+		TF2_AddCondition(this.iClient, TFCond_SpeedBuffAlly, this.flDuration, this.iClient);
+		this.flSpeed *= 1.2;
+		if(this.bSuperRage)
+		{
+			TF2_AddCondition(this.iClient, TFCond_TeleportedGlow, this.flDuration, this.iClient);
+			this.flSpeed *= 1.2;
+		}
+	}
+	
+	public void OnThink()
+	{
+		if (g_flBombProjectileEnd[this.iClient] == 0.0)
+			return;
+		
+		float flGameTime = GetGameTime();
+		if (flGameTime <= g_flBombProjectileEnd[this.iClient])
+		{
+			if (g_flBombProjectileNext[this.iClient] > flGameTime) return;
+		
+			if (this.bSuperRage)
+				g_flBombProjectileNext[this.iClient] = flGameTime + (this.flRate / 1.5);
+			else
+				g_flBombProjectileNext[this.iClient] = flGameTime + this.flRate;
+			
+			float vecOrigin[3], vecVelocity[3], vecAngleVelocity[3];
+			GetClientAbsOrigin(this.iClient, vecOrigin);
+			vecOrigin[2] += 42.0;
+			
+			for (int i = 0; i < 8; i++)
+			{
+				int iBomb = CreateEntityByName("tf_projectile_jar_gas");
+				if (iBomb > MaxClients)
+				{
+					//Create random velocity, but keep it upwards
+					switch (i)
+					{
+						case 0:
+						{
+							vecVelocity[0] = this.flMaxDistance;
+							vecVelocity[1] = this.flMaxDistance;
+						}
+						case 1:
+						{
+							vecVelocity[0] = -this.flMaxDistance;
+							vecVelocity[1] = this.flMaxDistance;
+						}
+						case 2:
+						{
+							vecVelocity[0] = this.flMaxDistance;
+							vecVelocity[1] = -this.flMaxDistance;
+						}
+						case 3:
+						{
+							vecVelocity[0] = -this.flMaxDistance;
+							vecVelocity[1] = -this.flMaxDistance;
+						}
+						case 4:
+						{
+							vecVelocity[0] = this.flMaxDistance;
+							vecVelocity[1] = 0.0;
+						}
+						case 5:
+						{
+							vecVelocity[0] = -this.flMaxDistance;
+							vecVelocity[1] = 0.0;
+						}
+						case 6:
+						{
+							vecVelocity[0] = 0.0;
+							vecVelocity[1] = this.flMaxDistance;
+						}
+						case 7:
+						{
+							vecVelocity[0] = 0.0;
+							vecVelocity[1] = -this.flMaxDistance;
+						}
```

Comment:
This whole hardcode should be able to be replaced by creating `45 * i` degree angle, then converting to velocity `GetAngleVectors`